rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDependency.java

```diff
@@ -258,6 +272,61 @@ public TreeVisitor<?, ExecutionContext> getVisitor(Scanned acc) {
                     private boolean isTopLevel(Cursor cursor) {
                         return cursor.getParentOrThrow().firstEnclosing(J.MethodInvocation.class) == null;
                     }
+
+                    private Predicate<Cursor> isMatchingJVMTestSuite(String resolvedConfiguration) {
+                        return cursor -> {
+                            String sourceSet = purgeConfigurationSuffix(resolvedConfiguration);
+                            J.MethodInvocation methodInvocation = cursor.getParentOrThrow().firstEnclosing(J.MethodInvocation.class);
+                            return methodInvocation != null && sourceSet.equals(methodInvocation.getSimpleName());
+                        };
+                    }
+
+                    private final Set<String> gradleStandardConfigurations = new HashSet<>(Arrays.asList(
+                            "api",
+                            "implementation",
+                            "compileOnly",
+                            "compileOnlyApi",
+                            "runtimeOnly",
+                            "testImplementation",
+                            "testCompileOnly",
+                            "testRuntimeOnly"));
+
+                    boolean targetsCustomJVMTestSuite(String configuration, Set<String> customJvmTestSuites) {
+                        if (gradleStandardConfigurations.contains(configuration) || "default".equals(configuration)) {
+                            return false;
+                        }
+
+                        String sourceSet = purgeConfigurationSuffix(configuration);
+                        return customJvmTestSuites.contains(sourceSet);
+                    }
+
+                    private String purgeConfigurationSuffix(String configuration) {
+                        if (configuration.endsWith("Implementation")) {
+                            return configuration.substring(0, configuration.length() - 14);
+                        } else if (configuration.endsWith("CompileOnly")) {
+                            return configuration.substring(0, configuration.length() - 11);
+                        } else if (configuration.endsWith("RuntimeOnly")) {
+                            return configuration.substring(0, configuration.length() - 11);
+                        } else if (configuration.endsWith("AnnotationProcessor")) {
+                            return configuration.substring(0, configuration.length() - 19);
+                        } else {
+                            return configuration;
+                        }
+                    }
+
+                    private String purgeSourceSet(@Nullable String configuration) {
+                        if (configuration == null || configuration.endsWith("Implementation")) {
```

Comment:
Historically, for SaaS we've needed to use `org.openrewrite.internal.StringUtils#isBlank` here. I don't recall if that's still the case or not.

@timtebeek, do you recall?

Reply:
Good to be consistent here indeed, just in case initialization is off.